app/main/posts/modify/post-data-editor.directive.js

```diff
@@ -415,19 +429,24 @@ function PostDataEditorController(
             }
             request.$promise.then(function (response) {
                 var success_message = (response.status && response.status === 'published') ? 'notify.post.save_success' : 'notify.post.save_success_review';
+
                 // Save the updated post back to outside context
-                $scope.postContainer.post = response;
+                $scope.postContainer.post = $scope.post;
```

Comment:
This definitely needs to remain as `response` otherwise we get empty fields showing up in the detail view and other weirdness.

Reply:
Reverted!